sceptre/stack.py

```diff
@@ -192,7 +197,22 @@ def update(self):
             "%s - Update stack response: %s", self.name, response
         )
 
-        status = self._wait_for_completion()
+        status = self._wait_for_completion(self.stack_timeout)
+        # Cancel update after timeout
+        if status == StackStatus.IN_PROGRESS:
```

Comment:
I think I would prefer if line 203 - 215 were consolidated into its own function; just to make things a bit neater.

Reply:
I'll create a cancel_stack_update method then.
Should the log line:
```python
        self.logger.warning(
            "%s - Update stack time exceeded the specified timeout",
            self.name
        )
```
be kept the same? It assumes the stack cancellation happened because of a timeout exceed - which might not always be the case if the method is reused in the future.

Reply:
Sounds good to me!